hibernate-reactive-core/src/main/java/org/hibernate/reactive/provider/service/ReactiveImprovedExtractionContextImpl.java

```diff
@@ -0,0 +1,93 @@
+/* Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright: Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.reactive.provider.service;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.concurrent.CompletionStage;
+
+import org.hibernate.boot.model.naming.Identifier;
+import org.hibernate.engine.jdbc.env.spi.JdbcEnvironment;
+import org.hibernate.reactive.pool.ReactiveConnection;
+import org.hibernate.reactive.pool.ReactiveConnectionPool;
+import org.hibernate.resource.transaction.spi.DdlTransactionIsolator;
+import org.hibernate.service.ServiceRegistry;
+import org.hibernate.tool.schema.internal.exec.ImprovedExtractionContextImpl;
+import org.hibernate.tool.schema.internal.exec.JdbcContext;
+
+import static org.hibernate.reactive.util.impl.CompletionStages.logSqlException;
+import static org.hibernate.reactive.util.impl.CompletionStages.returnOrRethrow;
+
+public class ReactiveImprovedExtractionContextImpl extends ImprovedExtractionContextImpl {
+
+	private final ReactiveConnectionPool service;
+
+	public ReactiveImprovedExtractionContextImpl(
+			ServiceRegistry registry,
+			Identifier defaultCatalog,
+			Identifier defaultSchema,
+			DatabaseObjectAccess databaseObjectAccess) {
+		super(
+				registry,
+				registry.getService( JdbcEnvironment.class ),
+				new DdlTransactionIsolator() {
+					@Override
+					public JdbcContext getJdbcContext() {
+						return null;
+					}
+
+					@Override
+					public void prepare() {
+					}
+
+					@Override
+					public Connection getIsolatedConnection() {
+						return null;
+					}
+
+					@Override
+					public void release() {
+					}
+				},
+				defaultCatalog,
+				defaultSchema,
+				databaseObjectAccess
+		);
+		service = registry.getService( ReactiveConnectionPool.class );
+	}
+
+	@Override
+	public <T> T getQueryResults(
+			String queryString,
+			Object[] positionalParameters,
+			ResultSetProcessor<T> resultSetProcessor) throws SQLException {
+
+		final CompletionStage<ReactiveConnection> connectionStage = service.getConnection();
```

Comment:
@gavinking, I noticed that [`ReactiveGenerationTarget#executeCommands`](https://github.com/hibernate/hibernate-reactive/blob/main/hibernate-reactive-core/src/main/java/org/hibernate/reactive/provider/service/ReactiveGenerationTarget.java#L91-L105) uses the same `ReactiveConnection` to execute all of the DDL statements.

Here, `Service#getConnection` gets called for each query that gets executed. That `ReactiveConnection` gets closed in the `finally` block below.

Would it be better to keep a `ReactiveConnection` around so the same one is used for all of the queries?

Reply:
I can't think of a reason why it would matter.

Reply:
OK, I'll leave it as is.